test/overseer/store/jdbc_test.clj

```diff
@@ -0,0 +1,113 @@
+(ns overseer.store.jdbc-test
+ (:require [clojure.test :refer :all]
+           [framed.std.core :as std]
+           loom.graph
+           (overseer
+             [api :as api]
+             [core :as core]
+             [test-utils :as test-utils])
+           [overseer.store.jdbc :as store]
+           [overseer.store-test :as store-test]))
+
+(defn- test-store
+  "Generate a fresh/configured store against a new in-memory H2 DB"
+  []
+  (let [store
+        (store/store
+          {:adapter :h2
+           :db-spec
+           {:classname "org.h2.Driver"
+            :subprotocol "h2:mem://"
+            :subname (str (std/rand-alphanumeric 12) ";DB_CLOSE_DELAY=-1")
+            :user "sa" ; System Administrator username
+            :password ""}})]
+    (core/install store)
+    store))
+
+(deftest test-where-map->clause
+  (is (= "age = 30 AND email = 'bob@example.com'"
+         (store/where-map->clause
+           {:age 30
+            :email "bob@example.com"}))))
+
+(deftest test-job->jdbc-map
+  (let [job {:job/id "foo-id"
+             :job/type "process"
+             :job/status :unstarted
+             :job/args {:customer_id 1}}]
+    (is (= {:id "foo-id"
+            :type "process"
+            :status "unstarted"
+            :args (pr-str {:customer_id 1})}
+           (store/job->jdbc-map job)))))
+
+(deftest test-jdbc-map->job
+  (let [jdbc-map {:id "foo-id"
+                  :type "process"
+                  :status "unstarted"
+                  :args (pr-str {:customer_id 1})}]
+    (is (= {:job/id "foo-id"
+            :job/type :process
+            :job/status :unstarted
+            :job/args {:customer_id 1}}
+           (store/jdbc-map->job jdbc-map)))))
+
+(deftest test-job-jdbc-map-roundtrip
+  (let [job (test-utils/job)]
+    (is (= job (store/jdbc-map->job (store/job->jdbc-map job))))))
+
+(deftest test-update-job
+  ; Test that optimistic concurrency is implemented correctly (via lock_version)
+  ; Start two concurrent update attempts on the same row; expect only 1 to win
+  ; and subsequently update the row / increment the lock
+  (let [{:keys [db-spec] :as store} (test-store)
+        {job-id :job/id :as job} (test-utils/job)
+
+        conflicts (atom 0)
+
+        attempt-update
+        (fn []
+          (try
+            (store/update-job db-spec job-id {:status "unstarted"} {:status "started"})
+            (catch clojure.lang.ExceptionInfo ex
+              (if (= :stale-object (:type (ex-data ex)))
+                (swap! conflicts inc)
+                (throw ex)))))]
+    (core/transact-graph store (api/simple-graph job))
+    (run! deref [(future (attempt-update))
```

Comment:
I'm not 100% sure how to do a real concurrent test but if you wanted to test it under more real concurrency conditions you'd want to do like an aphyr-style thing and really hammer it with a lot of threads (like queue up more threads than cores)--otherwise you can't tell if only 1 won _because the second thread queued/started after the first one finished_.